src/oic/utils/shelve_wrapper.py

```diff
@@ -16,31 +19,46 @@ def __len__(self):
         return db.__len__()
 
     def has_key(self, key):
+        if six.PY2 and not isinstance(key, six.string_types):
+            key = self._normalizeString(key)
         return key in self
 
     def __contains__(self, key):
+        if six.PY2 and not isinstance(key, six.string_types):
+            key = self._normalizeString(key)
         db = self._reopen_database()
         return db.__contains__(key)
 
     def get(self, key, default=None):
+        if six.PY2 and not isinstance(key, six.string_types):
+            key = self._normalizeString(key)
         db = self._reopen_database()
         return db.get(key, default)
 
     def __getitem__(self, key):
+        if six.PY2 and not isinstance(key, six.string_types):
+            key = self._normalizeString(key)
         db = self._reopen_database()
         return db.__getitem__(key)
 
     def __setitem__(self, key, value):
+        if six.PY2 and not isinstance(key, six.string_types):
+            key = self._normalizeString(key)
         db = self._reopen_database()
         db.__setitem__(key, value)
 
     def __delitem__(self, key):
+        if six.PY2 and not isinstance(key, six.string_types):
+            key = self._normalizeString(key)
         db = self._reopen_database()
         db.__delitem__(key)
 
     def _reopen_database(self):
         return shelve.open(self.filename, writeback=True)
 
+    def _normalizeString(self, key):
+        return unicodedata.normalize('NFKD', key).encode('ascii', 'ignore')
```

Comment:
Thanks for review! I defer to @VJalili on this one (see https://github.com/OpenIDC/pyoidc/pull/343#issuecomment-305991554 below).